compoments/report-repair/repair-availability.js

```diff
@@ -36,28 +49,28 @@ const RepairAvailability = ({handleChange, values, fromDate}) => {
     nextAppointmentSearchFromDate = moment.max(startTimes).add(1, 'day').format('YYYY-MM-DD');
 
     data.forEach((d) => {
-      const date = moment(d.startTime)
-      const dateString = date.format('Do MMMM YYYY')
-      const startTime = date.format('h:mma');
-      const endTime = moment(d.endTime).format('h:mma')
+      const startDateTime = moment(d.startTime)
+      const dateString = startDateTime.format('Do MMMM YYYY')
+      const startTime = startDateTime.format('h:mma');
+      const endDateTime = moment(d.endTime);
+      const endTime = endDateTime.format('h:mma')
+      const appointmentSlotKey = CreateAppointmentSlotKey(startDateTime, endDateTime)
       const timeString = `${startTime} to ${endTime}`
+      const timeStringSummary = `between ${timeString}`
+      const appointmentSlotData = {timeString, startDateTime:d.startTime, endDateTime:d.endTime, appointmentSlotKey}
       availability[dateString] ?
-        availability[dateString].push(timeString) :
-        availability[dateString] = [timeString]
+        availability[dateString].push(appointmentSlotData) :
+        availability[dateString] = [appointmentSlotData]
+      availabilityValues[appointmentSlotKey] = {startDateTime:d.startTime, endDateTime:d.endTime, display:`${dateString} ${timeStringSummary}`}
```

Comment:
The `availability` map is keyed by `dateString` to allow nesting a days slots within the UI.

The desired state changed by this component needs individual values for start and end times and that's what drove the creation of `availabilityValues`, which is keyed by `appointmentSlotKey`. 

And the `value` property of the radio buttons is also `appointmentSlotKey` allowing lookup upon change.

Happy to change if there is a better and/or simpler way.